src/main/java/interface_adaptors/AddController.java

```diff
@@ -17,8 +20,18 @@ public AddController(AddInputBoundary useCase){
      * @return a String for the View to present
      */
     public String addItem(String serialNum, int quantity){
+        if (quantity < 0){
+            return addPresenter.prepareFailure(0);
+        } // if the employee enters a negative quantity (unlikely, but you don't want this to happen at all)
+        else if (! addUseCase.serialNumberValid(serialNum)){
+            return addPresenter.prepareFailure(1);
+        } // can remove this later, assuming that serial number entered is never incorrect - just kept in for now for testing purposes
+
```

Comment:
Hey Tony,
I am a little confused about the view model being used here. I remember we reviewed this in the meeting but decided that the call to update the view should be made through the AddUseCase class to the presenter interface to implement the MVP model but looks like the use case is returning back to the controller and it doesn't coincide with any of the Model View patterns in clean architecture.

Reply:
Right, I've added changes so that the presenter calls an "IView", which is the interface for the screen.

Reply:
I think we should stick to the MVC model that is used in this implementation and instead just use a dependence inversion to ensure that we can stick to the MVC model. As this implementation has the most functionality, I think we would benefit from following this code style.